Software/Velvet_ESP32/lib/ESP32_STM_UART/src/ESP32_STM_UART.cpp

```diff
@@ -0,0 +1,258 @@
+ * * {0x02, 2B Size of FW, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду BOOTLOADER_READY_CMD, размер прошивки;
+ * * {0x03, 256B FW Data} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_PREPARE, передача данных прошивки;
+ * * {0x04, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_PREPARE, сообщает об успешной подготовке к передаче весовых данных.
+ * * {0x05, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_WEIGHT_ARRAY, сообщает об успешной передаче весовых данных.
+ * * {0x05, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_WEIGHT_ARRAY, сообщает об ошибке передачи весовых данных по WiFi.
+ * * {0x06, 4B ts – unix epoch, 0x00, 0x00, 0x00} - ответ на команду GET_TIMESTAMP_CMD, содержит запрошенную метку времени в формате unix timestamp.
+ * * {0x06, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду GET_TIMESTAMP_CMD, сообщает об ошибке при запросе метки времени.
+*/
+void ESP32_STM_UART::processIncomingMessage() {
+    if (!_serial.available()) {
+        return;
+    }
+    
+    uint8_t cmd = _serial.read();
+    
+    switch (cmd) {
+        case CHECK_FW_CMD: {
+            // Обработка CHECK_FW_CMD
+            uint8_t version0 = _serial.read();
+            uint8_t version1 = _serial.read();
+            checkFirmware(version0,version1);
+            break;
+        }
+        case BOOTLOADER_READY_CMD:{
+            sendFirmwareToSTM(_firmwareData, _firmwareSize);
+            break;
+        }
+        case SEND_WEIGHT_CMD_TRANSMIT_PREPARE: {
+            // Обработка SEND_WEIGHT_CMD_TRANSMIT_PREPARE
+            sendWeightCmdTransmitPrepare();
+            break;
+        }
+        case GET_TIMESTAMP_CMD: {
+            // Обработка GET_TIMESTAMP_CMD
+            uint32_t timestamp = getUnixTimestamp();           
+            if (timestamp != 0) {
+                sendTimestampRsp(timestamp, 0x00);
+            } else {
+                sendTimestampRsp(0, 0x02);
+            }
+            break;
+        }
+       
+    }
+}
+
+/**
+ * @brief Проверяет и получает обновление прошивки с сервера при наличии новой версии.
+ * 
+ * Функция проверяет доступность Wi-Fi, а затем отправляет запрос на сервер для получения обновления прошивки.
+ * Если сервер возвращает успешный ответ (HTTP-код 200), функция проверяет контрольную сумму (CRC32) полученного файла
+ * прошивки и сравнивает ее с контрольной суммой, полученной из заголовка ответа сервера. Если контрольные суммы
+ * совпадают, функция сохраняет данные прошивки и отправляет ответ на запрос проверки версии прошивки с результатом
+ * "Перезагрузить STM".
+ *
+ * @param version0 Младший байт версии прошивки, переданный из STM32.
+ * @param version1 Старший байт версии прошивки, переданный из STM32.
+ */
+void ESP32_STM_UART::checkFirmware(uint8_t version0, uint8_t version1) {
+    bool wi_fi = check_wifi();
+    if (!wi_fi){
```

Comment:
I think that would be better
```suggestion
    if (!check_wifi()){
```

Reply:
Yeah. You're right. Change completed.